### ISSUE TITLE
Required annotation fields not clear to users

### ISSUE THREAD

**Issue by kd-ods** [openownership/data-standard#333]
The [schema](https://github.com/openownership/data-standard/blob/master/schema/components.json#L394) sets out that `motivation` and `statementPointerTarget` are required within annotation objects. And if the motivation is 'linking' then `url` is also required.

However, the two fields that are always required are not picked up by the [tables in the docs](http://standard.openownership.org/en/0.2.0/schema/reference.html#annotation). I'm not sure how possible it is to fix up the tool that creates the tables from the schema to handle more complex requirements logic.

It's probably better to fix this by using the field descriptions to represent requirements.

**Comment by kd-ods** [comment]
This is linked to the validation issues in https://github.com/openownership/lib-cove-bods/issues/64.

It doesn't look like an alternative but equivalent logic can be used in the schema. Therefore, let's address this issue with edits to the property descriptions in the schema: describing which properties are required. (And moving to adopt the kind of [description style used in OCDS](https://ocds-standard-development-handbook.readthedocs.io/en/latest/meta/schema_style_guide.html#field-and-code-descriptions) is a good idea.)

**Comment by rhiaro** [comment]
Unfortunately "describing which properties are required" and "moving to adopt the kind of description style used in OCDS" are incompatible, as the OCDS style guide says:

> Descriptions should **not**:
> Explicitly state whether a field is required or optional.

**Comment by kd-ods** [comment]
Ah! Well, I understand why that would be part of the style guide: it is preferable that requirement is set at a schema level. I think we may need to ignore that element of the style guide in this case, since we can't communicate the requiredness(!) via the schema.

Or have you got a more elegant solution, @rhiaro? I don't *think* there's any reworking of the schema logic that will help, but you might know better than I do.

**Comment by rhiaro** [comment]
Similar issue with `Identifier`, where one or both `scheme` or `schemeName` are required, and the schema uses `anyOf` to express that. (That info is already in the description in that case though.)

I've opened an issue on the tool that generates the tables to dig into this further at that end.

**Comment by rhiaro** [comment]
I've reshuffled the schema logic so the two always required fields are at the top level of Annotation rather than in an `anyOf`. That means they show up correctly in the docs ([preview](https://standard.openownership.org/en/rhiaro-333-annotation/schema/reference.html#annotation)). I've adjusted the description for `url` since this is a conditional requirement and the tool that generates the tables can't handle that yet.
